frontend/alert/components/GroupSuggestion.tsx

```diff
@@ -0,0 +1,357 @@
+                <HistoryIcon icon={faHistory} className="historyIcon" />
+            </IconContainer>
+        </DropdownItemBox>
+    );
+};
+
+interface GroupSuggestionProps {
+    sections: any;
+    courseCode: string;
+    value: string;
+    selectedCourses: Set<Section>;
+    inputRef: React.RefObject<HTMLInputElement>;
+    setValue: (v: any) => void;
+    setTimeline: React.Dispatch<React.SetStateAction<string | null>>;
+    setSelectedCourses: any;
+}
+
+const GroupSuggestion = ({
+    sections,
+    courseCode,
+    value,
+    selectedCourses,
+    inputRef,
+    setValue,
+    setTimeline,
+    setSelectedCourses,
+}: GroupSuggestionProps) => {
+    const initializeButtonStates = () => {
+        const states = {};
+        Object.keys(sections).map((key) => {
+            states[key] = false;
+        });
+
+        return states;
+    };
+
+    const [buttonStates, setButtonStates] = useState(initializeButtonStates);
+
+    /**
+     * Takes the activity/activity of a section and return if all sections of {activity} have been selected
+     * Used to make sure to update button states when manually selecting each section
+     * @param activity - section activity
+     */
+    const selectedAllActivity = (activity) => {
+        // check if the user selected all courses of a certain activity in a group suggestion
+        let selectedAll = true;
+
+        sections[activity].map((section) => {
+            if (!selectedCourses.has(section)) {
+                selectedAll = false;
+            }
+        });
+
+        syncButtonStates(activity, selectedAll);
+        return selectedAll;
+    };
+
+    /**
+     * Update button states to match selected courses in the case the user is manually checking/unchecking sections
+     * Ex. allLectures should be true when user manually select all "LEC" sections
+     * @param activity - section activity
+     */
+    const syncButtonStates = (activity, correctState) => {
+        // update button state if its not the same as correctState
+        if (buttonStates[activity] != correctState) {
+            const newButtonStates = { ...buttonStates };
+            newButtonStates[activity] = correctState;
+            setButtonStates(newButtonStates);
+        }
+    };
+
+    /**
+     * Takes the activity of button clicked: LEC, REC, LABS, etc and switch state
+     * @param activity - section activity
+     */
+    const toggleButton = (activity) => {
+        const newButtonStates = { ...buttonStates };
+
+        newButtonStates[activity] = !newButtonStates[activity];
+        syncCheckAlls(activity, newButtonStates[activity], setSelectedCourses);
+
+        setButtonStates(newButtonStates);
+    };
+
+    /**
+     * If one of the select all is toggled, update selected courses to match
+     * @param activity - section activity
+     * @param checkedAll - whether all sections of activity should be checked or not
+     * @param setSelectedCourses - method to update selected courses set
```

Comment:
Any reason why this function needs to take `setSelectedCourses` as an argument, but `toggleButton` doesn't?

Reply:
Oops, I didn't need to pass `setSelectedCourses` into `syncCheckAlls`.